ui/apps/platform/src/Containers/Policies/Wizard/Step3/PolicyCriteriaOptions.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Button,
+    Divider,
+    Flex,
+    FlexItem,
+    Title,
+    Toolbar,
+    ToolbarContent,
+    ToolbarItem,
+    TreeView,
+    TreeViewDataItem,
+    TreeViewSearch,
+} from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { kebabCase } from 'lodash';
+import { useFormikContext } from 'formik';
+
+import { Policy } from 'types/policy.proto';
+import { Descriptor } from './policyCriteriaDescriptors';
+import { getEmptyPolicyFieldCard } from '../../policies.utils';
+
+import './PolicyCriteriaOptions.css';
+
+function getKeysByCategory(keys) {
+    const categories = {};
+    keys.forEach((key) => {
+        const { category } = key;
+        if (categories[category]) {
+            categories[category].push(key);
+        } else {
+            categories[category] = [key];
+        }
+    });
+    return categories;
+}
+
+function getPolicyFieldsAsTree(existingGroups, descriptors): TreeViewDataItem[] {
+    const categories = getKeysByCategory(descriptors);
+
+    const treeList = Object.keys(categories).map((category) => ({
+        name: '',
+        title: category,
+        id: kebabCase(category),
+        children: categories[category]
+            .filter((child) => {
+                const alreadyUsed = existingGroups.find(
+                    (group) =>
+                        group.fieldName.toLowerCase() === child?.name?.toLowerCase() ||
+                        group.fieldName.toLowerCase() === child?.label?.toLowerCase()
+                );
+                return !alreadyUsed;
+            })
+            .map<TreeViewDataItem>((child: Descriptor) => ({
+                name: child.longName,
+                title: child.shortName || child.name,
+                id: kebabCase(child.shortName),
+                icon: <PlusCircleIcon />,
+            })),
+    }));
+
+    return treeList;
+}
+
+type PolicyCriteriaOptionsProps = {
+    descriptors: Descriptor[];
+    selectedSectionIndex: number;
+};
+
+function PolicyCriteriaOptions({ descriptors, selectedSectionIndex }: PolicyCriteriaOptionsProps) {
+    // const [activeItems, setActiveItems] = useState<TreeViewDataItem[]>([]);
+    const [allExpanded, setAllExpanded] = useState(false);
+    const [filteredItems, setFilteredItems] = useState<TreeViewDataItem[]>([]);
+    const [isFiltered, setIsFiltered] = useState(false);
+    const { values, setFieldValue } = useFormikContext<Policy>();
+    const { policyGroups } = values.policySections[selectedSectionIndex] ?? [];
+
+    const selectedSection = values.policySections[selectedSectionIndex];
+
+    const treeDataItems = getPolicyFieldsAsTree([], descriptors);
+
+    useEffect(() => {
+        setFilteredItems(treeDataItems);
+        setIsFiltered(false);
+    }, [treeDataItems]);
```

Comment:
This is causing an infinite loop, as `treeDataItems` changes on each render.

Reply:
I also cannot seem to get the sidebar to filter, but that might be related to the effect loop here.